### DISCUSSION TITLE
Training data shuffle in Naive or other strategies?

### DISCUSSION THREAD

**Discussion by bartosz-grabowski** [ContinualAI/avalanche#1301] [Q&A] [answered]
Hi,

Is there data shuffling within each experience (not the experiences themselves) implemented for various strategies, particularly for `avalanche.training.supervised.Naive` strategy? I tried to find it but was not able to confirm whether the shuffling is being carried out during training.

**Comment by AntonioCarta** [comment, marked as answer]
Yes, there is and it's done by the `Dataloader`. By default, Avalanche uses a `TaskBalancedDataloader`, which shuffles and balance mini-batches by task. You can override it by setting `strategy.dataloader` with a plugin `before_training_exp`.

**Reply by bartosz-grabowski** [reply]
Thank you for your help.
